Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From h-shimamoto@ct.jp.nec.com Thu May 15 10:14:15 2008
Date: Mon, 28 Apr 2008 11:14:39 -0700
From: Hiroshi Shimamoto <h-shimamoto@ct.jp.nec.com>
To: Ingo Molnar <mingo@elte.hu>, Steven Rostedt <rostedt@goodmis.org>,
     Thomas Gleixner <tglx@linutronix.de>
Cc: linux-kernel@vger.kernel.org, linux-rt-users@vger.kernel.org
Subject: [PATCH -rt 1/4] x86_64: send NMI after nmi_show_regs on

From: Hiroshi Shimamoto <h-shimamoto@ct.jp.nec.com>

The flags nmi_show_regs should be set before send NMI.

Signed-off-by: Hiroshi Shimamoto <h-shimamoto@ct.jp.nec.com>
---
 arch/x86/kernel/nmi_64.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.25.4-rt2/arch/x86/kernel/nmi_64.c
===================================================================
--- linux-2.6.25.4-rt2.orig/arch/x86/kernel/nmi_64.c	2008-05-19 16:56:25.000000000 -0400
+++ linux-2.6.25.4-rt2/arch/x86/kernel/nmi_64.c	2008-05-19 16:56:35.000000000 -0400
@@ -324,11 +324,11 @@ void nmi_show_all_regs(void)
 	if (system_state == SYSTEM_BOOTING)
 		return;
 
-	smp_send_nmi_allbutself();
-
 	for_each_online_cpu(i)
 		nmi_show_regs[i] = 1;
 
+	smp_send_nmi_allbutself();
+
 	for_each_online_cpu(i) {
 		while (nmi_show_regs[i] == 1)
 			barrier();
